opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -165,9 +166,27 @@ def create(
         """
         if not attributes:
             attributes = {}
-        resource = _DEFAULT_RESOURCE.merge(
-            OTELResourceDetector().detect()
-        ).merge(Resource(attributes, schema_url))
+
+        resource = _DEFAULT_RESOURCE
+
+        for resource_detector in environ.get(
+            OTEL_EXPERIMENTAL_RESOURCE_DETECTORS, "otel"
+        ).split(","):
+
+            resource = resource.merge(
+                next(
+                    iter(
+                        entry_points(
+                            group="opentelemetry_resource_detector",
+                            name=resource_detector,
+                        )
+                    )
+                )
+                .load()()
+                .detect()
+            )
```

Comment:
Just load the resource detectors here and use `get_aggregated_resources`. It has exception and timeout handling not to block this indefinitely.

Reply:
Right, fixed.